projects/github-actions/test-results-to-slack/src/index.js

```diff
@@ -34,63 +35,66 @@ const { getMessage, sendMessage } = require( './slack' );
 	//endregion
 
 	const client = new WebClient( slackToken );
-
 	const isFailure = await isWorkflowFailed( ghToken );
 	const { text, id, mainMsgBlocks, detailsMsgBlocks } = await getNotificationData( isFailure );
-	const existingMessage = await getMessage( client, channel, id );
-	let mainMessageTS = existingMessage ? existingMessage.ts : undefined;
 	const icon_emoji = getInput( 'slack_icon_emoji' );
 
-	if ( existingMessage ) {
-		debug( 'Main message found' );
-		debug( 'Updating the main message' );
-		// Update the existing message
-		await sendMessage( client, true, {
-			text: `${ text }\n${ id }`,
-			blocks: mainMsgBlocks,
-			channel,
-			username,
-			ts: mainMessageTS,
-		} );
+	const channels = getChannels();
+	for ( const channel of channels ) {
```

Comment:
Seems like a good opportunity to move the contents of the for-loop into a separate folder. What do you think?

Reply:
That was my first attempt. I was not happy with it though, because it uses the constants set above, and I didn't like having another function with lots of params. I'll think about it some more with the next PRs. More logic will follow and I'll restructure all that code somehow.